src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -307,5 +318,35 @@ private static bool SupportsCSharpCodeActions(JToken token)
 
             return providesCodeActions && resolvesCodeActions;
         }
+
+        public override Task<JObject[]> UpdateConfigurationAsync(
+            OmniSharp.Extensions.LanguageServer.Protocol.Models.ConfigurationParams configParams,
+            CancellationToken cancellationToken)
+        {
+            var insertSpaces = _clientOptionsMonitor.InsertSpaces;
+            var tabSize = _clientOptionsMonitor.TabSize;
+
+            var result = new JObject[configParams.Items.Count()];
+            var index = 0;
+            foreach (var item in configParams.Items)
+            {
+                // Right now in VS we only care about editor settings, but we should update this logic later if
+                // we want to support Razor and HTML settings as well.
+                result[index] = item.Section == "editor"
+                    ? JObject.FromObject(new JsonEditorSettings { InsertSpaces = insertSpaces, TabSize = tabSize })
```

Comment:
At which point you wont need to do any of this JObject.FromObject stuff.

```C#
var setting = item.Section == "editor" ? new JsonEditorSettings { InsertSpaces = insertSpaces, TabSize = tabSize } : new obect();
result.Add(setting);
```

Normally I'd suggest factoring the settings out into a separate service but since this is a polyfill the quick way of doing this is fine